astropy/cosmology/connect.py

```diff
@@ -62,10 +79,10 @@ class CosmologyRead(io_registry.UnifiedReadWrite):
     -----
     """
 
-    def __init__(self, instance, cosmo_cls):
+    def __init__(self, instance: Cosmology, cosmo_cls: type[Cosmology]) -> None:
         super().__init__(instance, cosmo_cls, "read", registry=readwrite_registry)
 
-    def __call__(self, *args, **kwargs):
+    def __call__(self, *args: Any, **kwargs: Any) -> Cosmology:
```

Comment:
I just read the [blogpost](https://adamj.eu/tech/2021/05/07/python-type-hints-use-object-instead-of-any/)  linked to by @namurphy so I can't help but suggest to use `object` instead of `Any` to annotate `args` (for `kwargs` it's less clear and I think the "correct" annotation would be something like
```python
class CosmoKwargs(TypedDict, total=False):
    cosmology: str | type[Cosmology]
```
but I never remember the correct way to actually annotate `**kwargs` (I know it's not to be annotated as a dict), and honestly I'm not sure it's worth the effort of researching it. `Any` is probably fine here.

Reply:
Thank you for bringing this up! My current understanding is that `Any` should be used for dynamic typing (including if a method gets dynamically added to an `object) and `object` should be used when a static type checker would be able to keep track of how types get changed.  

Here's an example from the [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html) (which I should probably set as my homepage):
```python
# This says each positional arg and each keyword arg is a "str"
def call(self, *args: str, **kwargs: str) -> str: ...
```

Reply:
> The "correct" annotation would be something like
> 
> ```python
> class CosmoKwargs(TypedDict, total=False):
>     cosmology: str | type[Cosmology]
> ```

Unfortunately this means that cosmology is not required, not that other arguments are allowed. Python doesn't yet have a way to type a subset of kwargs. 

> but I never remember the correct way to actually annotate `**kwargs` (I know it's not to be annotated as a dict), and honestly I'm not sure it's worth the effort of researching it. `Any` is probably fine here.

The `Unpack` argument can be used.